lib/adobe/campaign/language.rb

```diff
@@ -0,0 +1,192 @@
+# frozen_string_literal: true
+module Adobe
+  module Campaign
+    class Language
+
+      def self.map_code(language_code)
+        return language_code if language_code.empty?
+
+        code = normalize language_code
+
+        return code if @adobe_language_codes.include?(code)
+        return code.gsub("-", "_") if @adobe_language_codes.include?(code.gsub("-", "_"))
+
+        if code.include?("-")
+          map_code code.rpartition("-").first
+        else
+          @adobe_language_codes.find{ |s| s.start_with? code}
+        end
+
```

Comment:
Code style concern: don't add new lines at the end of your method or block descriptions.